src/collections/dataloader.ts

```diff
@@ -51,63 +47,160 @@ const batchAndLoadDocs = (req: PayloadRequest): BatchLoadFn<string, TypeWithID>
     ];
 
     const batchKey = JSON.stringify(batchKeyArray);
-
-    const idField = payload.collections?.[collection].config.fields.find((field) => fieldAffectsData(field) && field.name === 'id');
-
-    if (isValidID(id, getIDType(idField))) {
-      return {
-        ...batches,
-        [batchKey]: [
-          ...batches[batchKey] || [],
-          id,
-        ],
-      };
-    }
+    let res: (value: TypeWithID) => void;
+    let rej: (reason?: any) => void;
+    results.push(new Promise<TypeWithID>((resolve, reject) => {
+      res = resolve;
+      rej = reject;
+    }));
+    // eslint-disable-next-line no-param-reassign
+    batches[batchKey] = [
+      ...batches[batchKey] || [],
+      {
+        id,
+        key,
+        resolve: res,
+        reject: rej,
+      },
+    ];
     return batches;
   }, {});
 
   // Run find requests in parallel
+  Object.entries(batchByFindArgs).forEach(async ([batchKey, ids]) => {
+    const allKeysInBatch = new Set<string>();
+    ids.forEach(({ key }) => allKeysInBatch.add(key));
+
+    try {
+      const [collection, depth, currentDepth, locale, fallbackLocale, overrideAccess, showHiddenFields] = JSON.parse(batchKey);
+
+      // copy req object to avoid mutating it
+      const request = { ...req };
```

Comment:
Are there other ways to copy a request?

Reply:
Unfortunately Node 15 introduces some issues with copying a `req`, and we had to work around that in our local operations. I think we might need to work around that here as well, otherwise the `req` that is passed to the `find` operation will lose properties in newer Node versions.

Take a look here:
https://github.com/payloadcms/payload/pull/975

Without being familiar with the exact reasons why we need to avoid mutating `req`, it's hard for me to understand the ramifications of not taking a copy of it.

What do you think?